src/megaclient.cpp

```diff
@@ -993,6 +999,73 @@ void MegaClient::exec()
 
         looprequested = false;
 
+        if (pendinghttp.size())
+        {
+            map<int, GenericHttpReq*>::iterator it = pendinghttp.begin();
+            while (it != pendinghttp.end())
+            {
+                GenericHttpReq *req = it->second;
+                switch (req->status)
+                {
+                case REQ_FAILURE:
+                    if (!req->httpstatus && (!req->maxretries || (req->numretry + 1) < req->maxretries))
+                    {
+                        req->numretry++;
+                        req->status = REQ_PREPARED;
```

Comment:
If a request is prepared for a retry, could be execute it without a new iteration of the `exec()` loop?